apps/system/js/app_chrome.js

```diff
@@ -515,14 +515,19 @@
     }
 
     var self = this;
-    var previousColor;
+    var finishedFade = false;
+    var endBackgroundFade = function() {
+      finishedFade = true;
+      self.element.removeEventListener('transitionend', endBackgroundFade);
+    };
+    this.element.addEventListener('transitionend', endBackgroundFade);
 
     window.requestAnimationFrame(function updateAppColor() {
```

Comment:
+1. If it's possible to detect when we cross the "light" brightness boundary then we should definitely use a setTimeout.